Fix one-off commands for "restart: unless-stopped" (fixes #6302)

hartwork opened the pull request:
Patch to fix #6302.

If this needs a dedicated automated test I'd welcome a pointer to get up to speed or you taking over, as you see fits.

Thanks for your consideration!

hartwork commented:
Fixed the CI now, happy to discuss if this is the best approach. Thanks!

shin- commented on compose/cli/main.py:
```diff
@@ -1327,6 +1328,9 @@ def run_one_off_container(container_options, project, service, options, toplevel
     project.initialize()
 
+    # Ensure that run command remains one-off (issue #6302)
+    container_options['restart'] = parse_restart_spec('no')
```
We can just set this to `None`, as no restart is the default. This also spares us the additional import.

hartwork commented:
Thanks for your review!

Since you mentioned that `if` branch to be removed, I had a closer look and come to the conclusion that `build_container_options` might be just the right place to do post-processing of the restart policy. So I moved the code there. I also renamed the method to `build_one_off_container_options` to better reflect it's scope, in a separate commit. If it's part of the public API, I'm happy to revert that rename.

What do you think?

PS: That Jenkins CI is very slow. If you have the power to make someone throw more resources at it, my vote for it.

hartwork commented:
Any chance this could go into the next release?

hartwork commented:
@shin- any chance this could be included with the next release?
